### DISCUSSION TITLE
`DatePicker` default value is string giving errors

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#47850] [Q&A]
Hello, when we pass a string value to a DatePicker, we get a `date.isValid is not a function` error, presumably because the default passed is a string and is not being converted to a dayjs object. These values come directly from our database call and we'd like a way for Datepicker components to automatically convert any string value it receives to a dayjs object. How can this be achieved directly with the DatePicker component?

**Comment by LovaArutinovi** [comment]
Hello, the same issue

**Comment by CooperHash** [comment]
At present, no. You should use day.js to wrap the default string.

**Comment by jessbraga** [comment]
Resolvi assim: 

import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import TimePicker from "antd/es/time-picker";
import dayjs from 'dayjs';
import type { Dayjs } from 'dayjs';

export function modal(props: string){

const formats = 'HH:mm';
const [hour, setHour] = useState<Dayjs | null>(dayjs(props.hour, formats));

return(
< Localization Provider dateAdapter={AdapterDayjs} >
   <TimePicker defaultValue={dayjs('12:08', formats)} format={formats} value={hour} onChange={onChange} disabled={readonlyDate}/>
</Localization Provider >
);
}

**Reply by jessbraga** [reply]
O correto é assim: LocalizationProvider. Escrevi separado porque não estava exibindo no comentário.
